pkg/build/model.go

```diff
@@ -0,0 +1,419 @@
+	if !filesystem.FileExistsAndNotDir(joinPath, fs) {
+		oktetoLog.Infof("Dockerfile '%s' is not in a relative path to context '%s'", b.Dockerfile, b.Context)
+		return b.Dockerfile
+	}
+
+	if joinPath != filepath.Clean(b.Dockerfile) && filesystem.FileExistsAndNotDir(b.Dockerfile, fs) {
+		oktetoLog.Infof("Two Dockerfiles discovered in both the root and context path, defaulting to '%s/%s'", b.Context, b.Dockerfile)
+	}
+
+	return joinPath
+}
+
+// AddBuildArgs add a set of args to the build information
+func (b *BuildInfo) AddBuildArgs(previousImageArgs map[string]string) error {
+	if err := b.expandManifestBuildArgs(previousImageArgs); err != nil {
+		return err
+	}
+	return b.addExpandedPreviousImageArgs(previousImageArgs)
+}
+
+func (b *BuildInfo) expandManifestBuildArgs(previousImageArgs map[string]string) (err error) {
+	for idx, arg := range b.Args {
+		if val, ok := previousImageArgs[arg.Name]; ok {
+			oktetoLog.Infof("overriding '%s' with the content of previous build", arg.Name)
+			arg.Value = val
+		}
+		arg.Value, err = env.ExpandEnv(arg.Value)
+		if err != nil {
+			return err
+		}
+		b.Args[idx] = arg
+	}
+	return nil
+}
+
+func (b *BuildInfo) addExpandedPreviousImageArgs(previousImageArgs map[string]string) error {
+	alreadyAddedArg := map[string]bool{}
+	for _, arg := range b.Args {
+		alreadyAddedArg[arg.Name] = true
+	}
+	for k, v := range previousImageArgs {
+		if _, ok := alreadyAddedArg[k]; ok {
+			continue
+		}
+		expandedValue, err := env.ExpandEnv(v)
+		if err != nil {
+			return err
+		}
+		b.Args = append(b.Args, BuildArg{
+			Name:  k,
+			Value: expandedValue,
+		})
+		oktetoLog.Infof("Added '%s' to build args", k)
+	}
+	return nil
+}
+
+// UnmarshalYAML Implements the Unmarshaler interface of the yaml pkg.
+func (e *BuildArg) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var raw string
+	err := unmarshal(&raw)
+	if err != nil {
+		return err
+	}
+	maxBuildArgsParts := 2
+
+	parts := strings.SplitN(raw, "=", maxBuildArgsParts)
+	e.Name = parts[0]
+	if len(parts) == maxBuildArgsParts {
+		e.Value = parts[1]
+		return nil
+	}
+
+	e.Name, err = env.ExpandEnv(parts[0])
+	if err != nil {
+		return err
+	}
+	e.Value = parts[0]
+	return nil
+}
+
+// UnmarshalYAML Implements the Unmarshaler interface of the yaml pkg.
+func (d *BuildDependsOn) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var rawString string
+	err := unmarshal(&rawString)
+	if err == nil {
+		*d = BuildDependsOn{rawString}
+		return nil
+	}
+
+	var rawStringList []string
+	err = unmarshal(&rawStringList)
+	if err == nil {
+		*d = rawStringList
+		return nil
+	}
+	return err
+}
+
+// UnmarshalYAML Implements the Unmarshaler interface of the yaml pkg.
+func (buildInfo *BuildInfo) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var rawString string
+	err := unmarshal(&rawString)
+	if err == nil {
+		buildInfo.Name = rawString
+		return nil
+	}
+
+	var rawBuildInfo buildInfoRaw
```

Comment:
NIT: Maybe we can do something like what we do [here](https://github.com/okteto/okteto/blob/master/pkg/model/serializer.go#L1257-L1273). In case we add or remove a new field it will be added directly and we won't be failing. WDYT?